### DISCUSSION TITLE
Pseudo ps2-era building destruction animation test.

### DISCUSSION THREAD

**Discussion by SniperGoth** [redeclipse/base#1146] [Maps and Content Showcase]
Just a little thing i did some weeks ago when i was learning how to export animations and configuring them on the engine...
You use the trigger, the animation plays, puff, no building, and only some small debrees are left on it's place.

![20171230141826](https://user-images.githubusercontent.com/24359960/34455739-688d0eea-ed6c-11e7-815c-6c6e55eea0c2.png)
![20171230141828](https://user-images.githubusercontent.com/24359960/34455740-68aeb388-ed6c-11e7-9659-9d4b8160aefa.png)
![20171230141830](https://user-images.githubusercontent.com/24359960/34455741-68cbc86a-ed6c-11e7-88cc-c3a47bc5b583.png)
![20171230141841](https://user-images.githubusercontent.com/24359960/34455742-68e893a0-ed6c-11e7-804f-4cbb07e2acb8.png)

https://youtu.be/8VSDE9gIEKo

If someone would like to test it by themselves to understand how it was made:
[destruction-testmap.zip](https://github.com/red-eclipse/world/files/1594542/destruction-testmap.zip)

**Comment by shirepirate** [comment]
that's pretty cool

**Comment by cloudziq** [comment]
thats working the same as every animated door in game, you can now make it better! ;p

**Comment by SniperGoth** [comment]
It's something that was in the back of my head for quite some time, this simple animation, hahah

**Comment by untrivializer** [comment]
> is simple

Did you mean, "_might be straightforward, but could have unforseen complexities that would completely change the prioritization of the issue, so we should let it go through the normal planning and estimation process_"[?](https://en.wikipedia.org/wiki/Planning_fallacy)

**Comment by acerspyro** [comment]
^ What the fuck

**Comment by qreeves** [comment]
LOL

**Comment by FabioOG** [comment]
Would it be possible to set some new trigger ent actived when there is an explosion to destroy the building ?

**Comment by SniperGoth** [comment]
In theory, if we had a projectile-activated trigger type, then one could maybe surround the building with the triggers, then be active when you shoot at it.
As for your suggestion, it would require so the trigger has some flags (projectile type : 1-bullet, 2-explosion) for it to work, otherwise, if a player claws or even does a sword attack around it, BOOM, building exploded.

**Comment by cloudziq** [comment]
this one of the basic feature from other games that I always missing in RE :/

**Comment by FabioOG** [comment]
Well, maybe instead of an action parameter we should make a mask parameter with all the weapons or action that active the trigger like proximity, pressing E, each weapon shot
